opensearch-project/OpenSearch-Dashboards
# [ignoreFilterIfFieldNotInIndex] Build separate Timeline filters by index

### Description

When ignoreFilterIfFieldNotInIndex is enabled:
1. Parse the Timeline visualization expression for indexes
2. For each index in the expression, build a separate OpenSearch query from the current filters
3. Pass the filters for each index to the Timeline request handler
4. In the request handler, use the index-specific query instead of the single index-agnostic query

### Issues Resolved

closes #6184

## Testing the changes

1. Enable ignoreFilterIfFieldNotInIndex 
2. Create dashboard with visualizations for more than one index
3. Create timeline visualization for one of those indexes and add it to the dashboard
4. Create filter that applies only to the index not referenced by the timeline visualization
5. The timeline visualization should no longer flatline, because the filter is no longer being applied to it

### Check List
